frame/election-provider-support/src/lib.rs

```diff
@@ -662,16 +656,166 @@ impl<AccountId: IdentifierT, Accuracy: PerThing128, Balancing: Get<Option<Balanc
 
 /// A voter, at the level of abstraction of this crate.
 pub type Voter<AccountId, Bound> = (AccountId, VoteWeight, BoundedVec<AccountId, Bound>);
-
 /// Same as [`Voter`], but parameterized by an [`ElectionDataProvider`].
 pub type VoterOf<D> =
 	Voter<<D as ElectionDataProvider>::AccountId, <D as ElectionDataProvider>::MaxVotesPerVoter>;
-
 /// Same as `BoundedSupports` but parameterized by a `ElectionProviderBase`.
 pub type BoundedSupportsOf<E> = BoundedSupports<
 	<E as ElectionProviderBase>::AccountId,
 	<E as ElectionProviderBase>::MaxWinners,
 >;
+/// Count bound of data provider bounds.
+pub type CountBound = u32;
+/// Size bound of data provider bounds.
+pub type SizeBound = u32;
 
 sp_core::generate_feature_enabled_macro!(runtime_benchmarks_enabled, feature = "runtime-benchmarks", $);
 sp_core::generate_feature_enabled_macro!(runtime_benchmarks_or_fuzz_enabled, any(feature = "runtime-benchmarks", feature = "fuzzing"), $);
+
+#[derive(Clone, Copy, Default, Debug)]
+pub struct DataProviderBounds {
+	pub count: Option<CountBound>,
```

Comment:
FYI if you make the newtype, these can just remain raw `u32`, because they are linked to a named filed that clears out what they are. 

The main hassle is when you have a function, and two consecitive arguments are u32.